aiosip/contact.py

```diff
@@ -8,28 +8,36 @@
 
 
 # Regex pattern from p2p-sip project
-CONTACT_PATTERNS = [re.compile('^(?P<name>[a-zA-Z0-9\-\._\+~ \t]*)<(?P<uri>[^>]+)>(?:;(?P<params>[^\?]*))?'),
+CONTACT_PATTERNS = [re.compile('^(?P<name>["a-zA-Z0-9\-\._\+~ \t]*)<(?P<uri>[^>]+)>(?:;(?P<params>[^\?]*))?'),
                     re.compile('^(?:"(?P<name>[^"]+)")[ \t]*<(?P<uri>[^>]+)>(?:;(?P<params>[^\?]*))?'),
                     re.compile('^[ \t]*(?P<name>)(?P<uri>[^;]+)(?:;(?P<params>[^\?]*))?')]
 
 
 class Contact(dict):
-    def __init__(self, contact):
-        for s in CONTACT_PATTERNS:
-            m = s.match(contact)
-            if m:
-                super().__init__(m.groupdict())
-                break
-        else:
-            raise ValueError('Not valid contact address')
-        if self['params']:
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        params = self.get('params')
+        if not isinstance(params, Param):
             self['params'] = Param(self['params'])
         else:
             self['params'] = Param()
+
         if 'tag' not in self['params']:
             self['params']['tag'] = gen_str(16, string.digits + 'abcdef')
-            contact_logger.debug('Contact: %s, params: %s', contact, self['params'])
-        self['uri'] = Uri(self['uri'])
+
+        uri = self.get('uri')
+        if not isinstance(uri, Uri):
+            self['uri'] = Uri(self['uri'])
+
+    @classmethod
+    def fromheader(cls, contact):
```

Comment:
I suggest to use from_header as class method name, to be consistent with the other methods and PEP8 itself.